### PR TITLE
Addressing issue #5182

### PR DESCRIPTION
**Pull request by Bezierr**
Added the options "Long Edge" and "Short Edge" to the Resize functionality.

### CODE REVIEW

File: rtdata/languages/default [hunk 1]
```diff
@@ -297,8 +297,8 @@ HISTORY_MSG_42;Exposure - Tone curve 2
 HISTORY_MSG_43;Exposure - Tone curve 2 mode
 HISTORY_MSG_44;Lum. denoising radius
 HISTORY_MSG_45;Lum. denoising edge tolerance
-HISTORY_MSG_46;Color denoising
-HISTORY_MSG_47;Blend ICC highlights with matrix
+HISTORY_MSG_46;Resize - Long Edge
```
Comment (Thanatomanic) [inline rtdata/languages/default:300]: What is going on here? You can't just change history messages...

File: rtengine/jdatasrc.cc [hunk 2]
```diff
@@ -3,6 +3,13 @@
...
 #include "jpeg.h"
 
+/*
+ * HA 2021-03-12
+ * Modification according to
+ * https://github.com/Beep6581/RawTherapee/commit/be67261d0bdeb21ffe15fe968dd5bc16edd407e0
+ * because otherwise it won't compile.
+ */
+  
 /*
```
Comment (Thanatomanic) [inline rtengine/jdatasrc.cc:13]: Where are the changes in this file coming from? These seem irrelevant to your PR.

File: rtgui/resize.cc [hunk 3]
```diff
@@ -81,19 +83,46 @@ Resize::Resize () : FoldableToolPanel(this, "resize", M("TP_RESIZE_LABEL"), fals
     Gtk::HBox* sbox = Gtk::manage (new Gtk::HBox ());
     Gtk::HBox* wbox = Gtk::manage (new Gtk::HBox ());
     Gtk::HBox* hbox = Gtk::manage (new Gtk::HBox ());
```
Comment (Thanatomanic) [inline rtgui/resize.cc:85]: This seems to be based on an older version of `resize.cc`. There have been changes https://github.com/Beep6581/RawTherapee/blob/53bb6e41bb1f4052bb85048fe72499cabcdcb1ec/rtgui/resize.cc#L91

Perhaps you need to rebase?

File: rtdata/languages/default (hunk 1 above)
Comment (Thanatomanic) [reply rtdata/languages/default:300]: Perhaps you can in this case since they are marked obsolete later on. Still, I would like to know for sure, just in case we break some backwards compatibility... If @heckflosse has time, he might want to comment.

**Review by Thanatomanic — CHANGES_REQUESTED** [review]
I have not tested the branch yet. There are some indications that you based this on a not-super-recent version of the `dev` branch (leading to merge conflicts). If you can look into that, that would be great. Another thing is the reuse of the history messages of which I have no clear idea if that is a good or bad idea...

File: rtengine/jdatasrc.cc (hunk 2 above)
Comment (Bezierr) [reply rtengine/jdatasrc.cc:13]: They are. But without them, it is apparently not possible to compile RT under Windows.

File: rtgui/resize.cc (hunk 3 above)
Comment (Bezierr) [reply rtgui/resize.cc:85]: It is in fact based on releases branch, not on dev branch, because the release version is what I use for actual raw processing. I'll try the rebasing as soon as I find the time.

File: rtdata/languages/default (hunk 1 above)
Comment (Bezierr) [reply rtdata/languages/default:300]: Not familiar with the works of the message mechanism, and the way you assign/reassign message numbers, reusing obsolete messages seemed to me the safest way to avoid breaking something. I think this could be changed fairly easily if you let me know what it should be.

File: rtengine/jdatasrc.cc (hunk 2 above)
Comment (Thanatomanic) [reply rtengine/jdatasrc.cc:13]: But the commit you refer to clearly shows these changes have been part of the codebase since October last year. You probably branched off earlier than that. This is not a problem per se, just inconvenient because it introduces merge conflicts. However, adding these comments to `jdatasrc.cc` makes no sense, they are completely irrelevant to include in the final code.

Comment (Bezierr) [reply rtengine/jdatasrc.cc:13]: Yes, releases branch is definitely older than that. That one, then, should also be resolved by a rebasing.

**Comment by Bezierr** [comment]
Will be replaced by a new Pull Request.

File: rtgui/resize.cc (hunk 3 above)
Comment (Beep6581) [reply rtgui/resize.cc:85]: @bezierr please base new work on the `dev` branch.
